presto-pinot/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java

```diff
@@ -0,0 +1,192 @@
+import com.google.inject.Inject;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.log.Logger;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.pinot.client.DynamicBrokerSelector;
+import org.apache.pinot.common.data.Schema;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static com.facebook.presto.pinot.PinotErrorCode.PINOT_FAILURE_GETTING_TABLE;
+import static com.facebook.presto.pinot.PinotUtils.TIME_COLUMN_NAME;
+import static com.facebook.presto.pinot.PinotUtils.TIME_COLUMN_VALUE;
+import static io.airlift.http.client.HttpStatus.familyForStatusCode;
+import static org.apache.pinot.common.config.TableNameBuilder.extractRawTableName;
+
+/**
+ * This class handles fetching necessary information from the Pinot cluster, including table schema, broker host, routing table, etc
+ */
+public class PinotClusterInfoFetcher
+        implements Closeable
+{
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final String GET_ALL_TABLES_API_TEMPLATE = "http://%s/tables";
+    private static final String TABLE_SCHEMA_API_TEMPLATE = "http://%s/tables/%s/schema";
+    private static final String ROUTING_TABLE_API_TEMPLATE = "http://%s/debug/routingTable/%s";
+    private static final String TIME_BOUNDARY_API_TEMPLATE = "http://%s/debug/timeBoundary/%s";
+    private static final CloseableHttpClient HTTP_CLIENT = HttpClients.createDefault();
+
+    private static final Logger log = Logger.get(PinotClusterInfoFetcher.class);
+
+    private final String controllerUrl;
+    private final String zookeeperServers;
+    private DynamicBrokerSelector dynamicBrokerSelector;
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    public PinotClusterInfoFetcher(PinotConfig pinotConfig)
+    {
+        this(pinotConfig.getZookeeperUrl(), pinotConfig.getPinotCluster(), pinotConfig.getControllerUrl());
+    }
+
+    public PinotClusterInfoFetcher(String zookeeperUrl, String pinotCluster, String controllerUrl)
+    {
+        log.info("Trying to init PinotClusterInfoFetcher with Zookeeper: %s, PinotCluster %s, ControllerUrl: %s.", zookeeperUrl, pinotCluster, controllerUrl);
+        zookeeperServers = zookeeperUrl + "/" + pinotCluster;
+        this.controllerUrl = controllerUrl;
+    }
+
+    public static String sendHttpGet(final String url)
+            throws Exception
+    {
+        HttpUriRequest request = RequestBuilder.get(url).setHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON).build();
```

Comment:
I suggest using the Airlift Http client (to be consistent with the rest of the project)